src/bio.c

```diff
@@ -278,6 +293,158 @@ void *bioProcessBackgroundJobs(void *arg) {
             if (job_type == BIO_CLOSE_AOF) close(job->fd_args.fd);
         } else if (job_type == BIO_LAZY_FREE) {
             job->free_args.free_fn(job->free_args.free_args);
+        } else if (job_type == BIO_SAVE_TO_DISK) {
+            int usemark;
+            char lastbytes[RDB_EOF_MARK_SIZE], buf[PROTO_IOBUF_LEN], eofmark[RDB_EOF_MARK_SIZE];
+            ssize_t readlen, nread, nwritten;
+            off_t left = 0, repl_transfer_last_fsync_off = 0;
+            int error = 0, main_thread_abort = 0, received_ping = 0;
+            connection *conn = job->save_to_disk_args.conn;
+            int dest_fd = job->save_to_disk_args.dest_fd;
+
+            /* Put the socket in blocking mode to simplify RDB transfer.
+             * We'll restore it when the RDB is received. */
+            connBlock(conn);
+            connRecvTimeout(conn, server.repl_syncio_timeout * 1000);
+            do {
+                received_ping = 0;
+                if (shouldAbortSave()) {
+                    main_thread_abort = 1;
+                    goto done;
+                }
+                nread = connSyncReadLine(conn, buf, 1024, server.repl_syncio_timeout * 1000);
+                if (nread == -1) {
+                    replicaBioSaveServerLog(LL_WARNING, "I/O error reading bulk count from PRIMARY: %s", connGetLastError(conn));
+                    error = 1;
+                    goto done;
+                } else {
```

Comment:
This `else` isn't necessary. remove for better readability.  This applies to other places in the code.